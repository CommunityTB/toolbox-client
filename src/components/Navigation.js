import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../assets/css/styles.css';
import logo from '../assets/images/ct_logo_color.svg';
import TokenService from '../services/token-service'
import history from '../history';

export default class Navigation extends Component {

    handleUserLogout = () => {
        TokenService.clearAuthToken()
        history.push('/')
    }
    
    renderLoginLinks = () => {
        return (
            <>
                {/* <div class="hamburger">
                    <button class="hamburger"></button>
                    <button class="cross"></button>
                </div> */}
                <ul className='nav_not_logged_in'>
                    <Link
                        to='/login'>
                        Login    
                    </Link>
                    <Link
                        className='link_space' 
                        to='/register'>
                        Register
                    </Link>
                    <Link 
                        className='link_space'
                        to='/inventory'>
                        Inventory    
                    </Link>
                    <Link
                        className='link_space' 
                        to='/about'>
                        About
                    </Link>
                </ul>
            </>
        )
    }

    renderLogoutLinks = () => {
        return (
            <div className='nav_logged_in'>
                <Link to='/'
                        onClick={this.handleUserLogout}
                        >
                        Logout
                </Link>
            </div>
        )
    }

    render() {
       
        return (
            <nav className ='nav'>
                <div><Link to='/'><img src={logo} className='logo' alt='Community Toolbox Logo'></img></Link></div>
                <ul className='nav_items'>
                    { TokenService.hasAuthToken() ? this.renderLogoutLinks() : this.renderLoginLinks() }
                </ul>
            </nav>
        )
    }
}