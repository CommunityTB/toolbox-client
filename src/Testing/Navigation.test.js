import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Navigation from '../components/Navigation'

describe(`Navigation component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Navigation /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a Navigation by default', () => {
    const wrapper = shallow(<Navigation />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})