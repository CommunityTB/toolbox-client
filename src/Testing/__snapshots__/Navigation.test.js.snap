// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders a Navigation by default 1`] = `
<nav
  className="nav"
>
  <div>
    <Link
      to="/"
    >
      <img
        alt="Community Toolbox Logo"
        className="logo"
        src="ct_logo_color.svg"
      />
    </Link>
  </div>
  <ul
    className="nav_not_logged_in"
  >
    <Link
      to="/login"
    >
      Login
    </Link>
    <Link
      className="link_space"
      to="/register"
    >
      Register
    </Link>
    <Link
      className="link_space"
      to="/about"
    >
      About
    </Link>
  </ul>
</nav>
`;
